axlearn/common/attention_test.py

```diff
@@ -5091,6 +5091,70 @@ def test_set_double_shard_weights_config_for_list_of_configs(
                     (fsdp_axis_names, tp_axis_names, None),
                 )
 
+    @parameterized.product(
+    self_attention_input_linear_cfg=(
+        QKVLinear.default_config(),
+        FusedQKVLinear.default_config(),
+    ),
+    cross_attention_cfg=(None, TransformerAttentionLayer.default_config()),
+    batch_axis_names=("data", ("replica", "data", "fsdp")),
+    fsdp_axis_names=("fsdp",),
+    tp_axis_names=("model",),
+        seq_axis_names=("seq",),
+    )
+    def test_set_activation_shardings_config_for_list_of_configs(
+        self,
+        self_attention_input_linear_cfg,
+        cross_attention_cfg,
+        batch_axis_names,
+        fsdp_axis_names,
+        tp_axis_names,
+        seq_axis_names,
+    ):
+        cfg_layer: TransformerLayer.Config = TransformerLayer.default_config().set(
+            cross_attention=cross_attention_cfg
+        )
+        cfg_layer.self_attention.structure = "prenorm"
+        cfg_layer.feed_forward.structure = "prenorm"
+        cfg_layer.self_attention.attention.input_linear = self_attention_input_linear_cfg
+        cfg_layers = [cfg_layer, cfg_layer]
+
+        cfg_layer.self_attention.prenorm_partition_spec = (fsdp_axis_names, tp_axis_names, None,)
```

Comment:
interesting, is it the real use case you are testing nowadays?

Reply:
Yes, this is the real config for sequence parallel. I have confirmed with these specs and internal Neuron changes we can generate partitioned HLOs that perform Norms/Residuals/Dropouts partitioned along the sequence and matmuls partitioned along the model dimension without any unnecessary collectives, such as all-to-alls or collective-permutes or involuntary full rematerialization.